pkg/deployer/lib/continuousreconcile/extension.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package continuousreconcile
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/robfig/cron/v3"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	duv1alpha1 "github.com/gardener/landscaper/apis/deployer/utils/continuousreconcile/v1alpha1"
+	"github.com/gardener/landscaper/pkg/deployer/lib/extension"
+)
+
+// ContinuousReconcileExtension returns an extension hook function which can handle continuous reconciliation.
+// It is meant to be used as a ShouldReconcile hook and might yield unexpected results if used with another hook handle.
+// The function which it takes as an argument is expected to take a time and return the time when the deploy item should be scheduled for the next automatic reconciliation.
+//   It should return nil if continuous reconciliation is not configured for the deploy item.
+func ContinuousReconcileExtension(nextReconcile func(context.Context, time.Time, *lsv1alpha1.DeployItem) (*time.Time, error)) extension.ReconcileExtensionHook {
+	return func(ctx context.Context, log logr.Logger, di *lsv1alpha1.DeployItem, target *lsv1alpha1.Target, hype extension.HookType) (*extension.HookResult, error) {
+		logger := log.WithName("continuousReconcileExtension")
+		logger.V(7).Info("execute")
+		if di == nil {
```

Comment:
how can this happen?

Reply:
It's more of a safety measure, it can't happen unless someone uses the extension in a wrong way, I guess.

Reply:
ok but if it happens we should return an error then?
otherwise it would just silently not be called.
Maybe even a panic is better

Reply:
Good point 🤔

Reply:
I changed it to return an error in this case.

Reply:
Update: I changed it to a panic.